### ISSUE TITLE
Automatic rcon on local server.

### ISSUE THREAD

**Issue by AoeTW** [ddnet/ddnet#9009]
I've been mapping a lot lately and it is very tedious to always login to rcon. This past month I've logged in at least 500 times to test my map. Would be very nice if it was automatically logged in as admin as I join the server. 

Also another suggestion would be to make /practice always active. So you can instantly use your /tp bind to teleport as you join the server without having to join a team and then using /practice.

I know having practice on at all times is not something everyone wants. So maybe make it a command that you can add to autoexec_server.cfg like "sv_practice 1" or have it so that as long as you dont use /tp you will still be able to get a rank.

Third suggestion is being able to change the map by just typing : "map mapname" in rcon insted of using: "change_map mapname"

**Comment by sjrc6** [comment]
You can set the rcon password for your server in the config and then make a bind to log in with 1 key press.

**Comment by dropalways** [comment]
> You can set the rcon password for your server in the config and then make a bind to log in with 1 key press.

you can also use run on join

**Comment by AoeTW** [comment]
> You can set the rcon password for your server in the config and then make a bind to log in with 1 key press.

Yeah I know about that but I already filled out my entire keyboard with binds and started binding stuff with ctrl+key. My suggestions are really just quality of life changes to make it more convenient for mappers.

**Comment by dropalways** [comment]
> Also another suggestion would be to make /practice always active. So you can instantly use your /tp bind to teleport as you join the server without having to join a team and then using /practice.
> 

just make a bind to join team then vote practice?
`say /team 9;say /practice`

**Comment by dropalways** [comment]
> > You can set the rcon password for your server in the config and then make a bind to log in with 1 key press.
> 
> Yeah I know about that but I already filled out my entire keyboard with binds and started binding stuff with ctrl+key. My suggestions are really just quality of life changes to make it more convenient for mappers.

have you tried using tater client it has an option called bind wheel where u can have alot of commands activate with 1 key
![182528667-cf5bfbbf-ec10-4491-8c1b-21ff3b949715](https://github.com/user-attachments/assets/c21239ab-8b43-4311-92f3-02d2847828b9)

**Comment by AoeTW** [comment]
> > Also another suggestion would be to make /practice always active. So you can instantly use your /tp bind to teleport as you join the server without having to join a team and then using /practice.
> 
> just make a bind to join team then vote practice? `say /team 9;say /practice`

I already got those binds. Like I said, its all qol.

It is annoying to press every time. Also if you have a friend testing with you it is really annoying to wait for him to use /practice before you can teleport and joining team and using /practice with dummy. Also noticed that /practice doesn't work if you've already used an rcon command. So you need to kill, rejoin the team. /practice and then you can teleport.

**Comment by BlaiZephyr** [comment]
i dont see why not - automatically authing when joining localhost seems reasonable - +1

**Comment by ChillerDragon** [comment]
I already [proposed address based binds](https://github.com/ddnet/ddnet/pull/3552) 

``eval_if ui_server_address == "localhost" "rcon_auth rcon" else "rcon_auth strong_password"``

And the feature of a proper [password manager](https://github.com/ddnet/ddnet/issues/7633). Both proposals were rejected.

I use [chillerbot-ux](https://chillerbot.github.io/) for logging in on localhost/kog/block/city. It allows running custom commands such as ``rcon_auth`` based on the server ip automatically on join.

**Comment by heinrich5991** [comment]
This can probably be added easily once we get accounts.
